src/components/graphs/ContextGraph/ContextGraph.js

```diff
@@ -0,0 +1,130 @@
+    // `componentDidUpdate`.
+    getMetadata: PropTypes.func,
+    // `getMetadata` returns the metadata describing the datasets to
+    // be displayed in this component.
+    // A different function is passed by different clients to specialize
+    // this general component to particular cases (single vs. dual controller).
+    dataToGraphSpec: PropTypes.func,
+    // `dataToGraphSpec` converts data a graph spec.
+    // A different function is passed by different clients to specialize
+    // this general component to particular cases (single vs. dual controller).
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      graphSpec: blankGraphSpec,
+    };
+  }
+
+  displayNoDataMessage = (message) => {
+    //Removes all data from the graph and displays a message
+    this.setState({
+      graphSpec: noDataMessageGraphSpec(message),
+    });
+  };
+
+  getAndValidateTimeseries(metadata) {
+    return (
+      getData(metadata)
+        .then(validateLongTermAverageData)
+        .then(response => response.data)
+    );
+  }
+
+  loadGraph() {
+    this.displayNoDataMessage('Loading Data');
+
+    const metadatas = this.props.getMetadata().filter(metadata => !!metadata);
+    const timeseriesPromises = metadatas.map(metadata =>
+      this.getAndValidateTimeseries(metadata)
+    );
+
+    Promise.all(timeseriesPromises).then(data => {
+      this.metadatas = metadatas;
+      this.data = data;
+      this.setState({
+        graphSpec: this.props.dataToGraphSpec(metadatas, data, this.props.model_id),
+      });
+    }).catch(error => {
+      displayError(error, this.displayNoDataMessage);
+    });
+  }
+
+  // Lifecycle hooks
+
+  componentDidMount() {
+    this.loadGraph();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // If we are already displaying data, and if only the model_id has
+    // changed, then we don't need to get data again; we only need to
+    // regenerate the graph from it (to emphasize the current model).
+    // This code coordinates with `this.props.getMetadata`, and so may be
+    // better defined externally and passed as a prop (`updatedGraphSpec`,
+    // returning new graph spec?).
+    // Leave it here for now.
+    if (
+      this.state.graphSpec &&
+      this.state.graphSpec.data.columns.length > 0 &&
+      nextProps.variable_id === this.props.variable_id &&
+      nextProps.experiment === this.props.experiment
+    ) {
+      this.setState({
+        graphSpec: this.props.dataToGraphSpec(
+          this.metadatas, this.data, this.props.model_id),
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      // Note omission of model_id: This graph doesn't need to reload if model
+      // doesn't change. See `componentWillReceiveProps`.
+      prevProps.variable_id !== this.props.variable_id ||
+      prevProps.experiment !== this.props.experiment ||
+      prevProps.contextMeta !== this.props.contextMeta ||  // TODO: Necessary?
```

Comment:
I think currently the only circumstance in which `contextMeta` changes would be when the ensemble does, since `contextMeta` is currently just "all available dataset metadata" - the full results of the `multimeta` query.

... which, now that I think about it, doesn't seem ideal. No reason to pass` DataController` a bunch of file metadata it isn't going to use. contextMeta should probably be filtered to all datasets with the selected` variable_id` and `experiment` by `AppController`, and change when the user picks different values for `variable_id` or `experiment`. In that case, it would change when `ensemble`, `variable_id`, or `experiment` did, but not otherwise. That change is probably outside the scope of this PR, though.

Reply:
Added a new issue for this.